persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/util/CacheMapUtil.java

```diff
@@ -89,11 +89,17 @@
     }
 
     @PostConstruct
-    private void initializeCacheMemory() {
+    private void init() {
+        initializeCacheMemory();
+    }
+
+    public void initializeCacheMemory() {
+        
         // CHANGES TO THIS LIST MUST BE PROPAGATED TO 'GlobalProperties'
         this.cacheEnabled = (!authenticate.isEmpty() 
                 && !authenticate.equals("false") 
                 && !authenticate.contains("social_auth"));
+        
```

Comment:
This comment is actually about the checks farther down that look like this:
```java
        if (molecularProfileCache == null) {
            molecularProfileCache = new HashMap<String, MolecularProfile>();
        }
```

Why do we need those? Shouldn't we always be clearing the map prior to populating? Seems to me that this setup lets deleted studies remain in the cache even after it's cleared. I'd just always make a new map.

Reply:
yes, good point. Updated